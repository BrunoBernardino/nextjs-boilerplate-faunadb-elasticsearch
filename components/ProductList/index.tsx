import React from 'react';
import styled from 'styled-components';

import Product from 'components/Product';
import * as T from 'lib/types';

type ProductListProps = {
  products: T.Product[];
  className?: string;
};

const StyledProductList = styled.section`
  margin: 1em auto;
  padding: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
`;

const ProductList = (props: ProductListProps) => {
  const { products, className } = props;

  return (
    <StyledProductList className={className}>
      {products.map((product) => (
        <Product key={product.slug} {...product} />
      ))}
    </StyledProductList>
  );
};

ProductList.defaultProps = {
  className: '',
};

export default ProductList;
