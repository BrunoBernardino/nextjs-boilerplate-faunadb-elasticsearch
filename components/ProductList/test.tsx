import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import expect from 'expect';
import enzymify from 'expect-enzyme';

import ProductList from './index';

expect.extend(enzymify());

const defaultProps = {
  products: [
    {
      name: 'Product 1',
      slug: 'product-1',
      price: 15,
      views: 0,
    },
  ],
};

describe('ProductList', () => {
  it('renders the list without errors', () => {
    const wrapper = shallow(<ProductList {...defaultProps} />);
    expect(wrapper.first().text()).toBe('<Product />');
  });

  it('renders the list as expected', () => {
    const tree = renderer.create(<ProductList {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
