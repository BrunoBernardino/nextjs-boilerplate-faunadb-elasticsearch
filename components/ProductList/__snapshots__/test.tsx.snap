// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders the list as expected 1`] = `
<section
  className="ProductList__StyledProductList-sc-99squt-0 fWWnqk"
>
  <article
    className="Product__StyledProduct-sc-1e37ctb-0 gEmVZS"
  >
    <a
      href="/products/product-1"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <h3
        className="Product__Title-sc-1e37ctb-1 kYMdyh main__title"
      >
        Product 1
      </h3>
      <p
        className="Product__Paragraph-sc-1e37ctb-2 jkqygI main__description"
      >
        Price: 
        <strong>
          $
          15
        </strong>
      </p>
      <p
        className="Product__Paragraph-sc-1e37ctb-2 jkqygI main__description"
      >
        Views: 
        <strong>
          0
        </strong>
      </p>
    </a>
  </article>
</section>
`;
